libcontainer/integration/exec_test.go

```diff
@@ -1992,3 +1992,70 @@ next_fd:
 		t.Fatalf("found %d extra fds after container.Run", count)
 	}
 }
+
+// Test that a container using user namespaces is able to bind mount a folder
+// that has some part of its path with "---" permissions for others
+func TestBindMountAndUser(t *testing.T) {
+	if _, err := os.Stat("/proc/self/ns/user"); os.IsNotExist(err) {
```

Comment:
nit: new code should use `errors.Is(err, os.ErrNotExist)`.

Reply:
(I realize there's a bunch of other tests that use `os.IsNotExist` so feel free to ignore this)

Reply:
No problem, I've changed it :)